app/code/Magento/CatalogInventory/Helper/ProductStockIsQtyDecimal.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Magento\CatalogInventory\Helper;
+
+use Magento\CatalogInventory\Api\StockItemRepositoryInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class GetProductIsQtyDecimalService
+ */
+class ProductStockIsQtyDecimal
+{
+    /**
+     * @var StockItemRepositoryInterface
+     */
+    protected $stockItemRepository;
+
+    /**
+     * @param StockItemRepositoryInterface $stockItemRepository
+     *
+     * @return void
+     */
+    public function __construct(StockItemRepositoryInterface $stockItemRepository)
+    {
+        $this->stockItemRepository = $stockItemRepository;
+    }
+
+    /**
+     * @param $productId
+     *
+     * @return bool
+     */
+    public function execute($productId)
+    {
+        try {
+            $productStock = $this->stockItemRepository->get($productId);
+
+            return (bool) $productStock->getIsQtyDecimal();
+        } catch (NoSuchEntityException $e) {
+            return false;
```

Comment:
Please consider looking for another way to implement this, avoiding exception-managed business logic.